### PR TITLE
Indent-based folding and matching for Visual Basic Mode

### PR DESCRIPTION
**Pull request by mkslanc**
*Issue #, if available:*

*Description of changes:*
Added indent-based folding and tag matching for Visual Basic Mode. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/ace/mode/vbscript.js [hunk 1]
```diff
@@ -43,17 +43,160 @@ define(function(require, exports, module) {
...
+
+        var tokens = this.getTokenizer().getLineTokens(line.trim(), state).tokens;
+
+        if (!tokens || !tokens.length)
+            return false;
+        var val = tokens[0].value.toLowerCase();
+        return ((tokens[0].type == "keyword.control.asp" || tokens[0].type == "storage.type.function.asp") && outdentKeywords.indexOf(val) != -1);
+    };
+
+    this.getMatching = function(session, row, column, tokenRange = true) {
```
Comment (nightwing) [inline lib/ace/mode/vbscript.js:170]: Could you please convert this to es5 style, not using default arguments, we still do not use es6 to support older browsers.

File: lib/ace/mode/vbscript.js [hunk 2]
```diff
@@ -43,17 +43,160 @@ define(function(require, exports, module) {
...
 };
 oop.inherits(Mode, TextMode);
 
 (function() {
-       
+
     this.lineCommentStart = ["'", "REM"];
-    
+
+    var indentKeywords = {
```
Comment (nightwing) [inline lib/ace/mode/vbscript.js:60]: is it possible to reuse the value from folding rules for this?

File: lib/ace/mode/folding/vbscript.js [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+        if (endRow > startRow) {
+            var endColumn = session.getLine(endRow).length;
+            return new Range(startRow, startColumn, endRow, endColumn);
+        }
+    };
+
+    // must return "" if there's no fold, to enable caching
+    this.getFoldWidget = function(session, foldStyle, row) {
+        var line = session.getLine(row);
```
Comment (nightwing) [inline lib/ace/mode/folding/vbscript.js:98]: If you have code to detect blocks based on syntax, why not use it for fold widgets too?

something like 

```js
    this.getFoldWidget = function(session, foldStyle, row) {
        var line = session.getLine(row);
        var match = /^\s*(\w+)\b/i.exec(line)
        var keyword = match && match[1].toLowerCase();
        
        if (keyword && this.indentKeywords.hasOwnProperty(keyword)) {
            if (this.indentKeywords[keyword] == -1)
                return "";
            if (keyword == "if" && !/then\s*$/i.test(line))
                return ""
            return "start";
        }
        
        return "";
    };
```

File: lib/ace/mode/vbscript.js (hunk 1 above)
Comment (mkslanc) [reply lib/ace/mode/vbscript.js:170]: Oh, sure i could

File: lib/ace/mode/folding/vbscript.js (hunk 3 above)
Comment (mkslanc) [reply lib/ace/mode/folding/vbscript.js:98]: Yeah, i think i could handle block-based folding and matching

**Pull request merged**